### PR TITLE
[v2] Merge July 27th main into v2 feature branch

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

Merges main into the v2 feature branch.

## Why is it important?

So changes stay synced.

### CODE REVIEW

**Comment by blakerouse** [comment]
I need to run `go mod tidy` on this PR, but I cannot run it successfully until #790 is merged.

File: internal/pkg/agent/application/upgrade/upgrade.go [hunk 1]
```diff
@@ -84,9 +80,19 @@ func (u *Upgrader) Upgrade(ctx context.Context, version string, sourceURI string
 	span, ctx := apm.StartSpan(ctx, "upgrade", "app.internal")
 	defer span.End()
 
+	err = preUpgradeCleanup(u.agentInfo.Version())
```
[the remark this answers is not in the document]
Comment (blakerouse) [reply internal/pkg/agent/application/upgrade/upgrade.go:119]: Fixed it.

File: internal/pkg/agent/application/upgrade/upgrade.go [hunk 2]
```diff
@@ -84,9 +80,19 @@ func (u *Upgrader) Upgrade(ctx context.Context, version string, sourceURI string
...
+	if err != nil {
+		u.log.Errorf("Unable to clean downloads dir %q before update: %v", paths.Downloads(), err)
+	}
+
 	sourceURI = u.sourceURI(sourceURI)
 	archivePath, err := u.downloadArtifact(ctx, version, sourceURI)
 	if err != nil {
+		// Run the same preUpgradeCleanup task to get rid of any newly downloaded files
+		// This may have an issue if users are upgrading to the same version number.
+		if dErr := preUpgradeCleanup(u.agentInfo.Version()); dErr != nil {
```
[the remark this answers is not in the document]
Comment (blakerouse) [reply internal/pkg/agent/application/upgrade/upgrade.go:129]: Fixed it.

File: internal/pkg/agent/cmd/diagnostics_test.go [hunk 3]
```diff
@@ -137,3 +138,76 @@ func Test_collectEndpointSecurityLogs_noEndpointSecurity(t *testing.T) {
...
+				},
+			},
+			wantRedacted: []string{
+				"some_key", "a_password", "my_token", "4242key", "4242password", "4242certificate"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := redact(tt.arg)
```
[the remark this answers is not in the document]
Comment (blakerouse) [reply internal/pkg/agent/cmd/diagnostics_test.go:193]: Added to the commented out section for now.

File: internal/pkg/agent/transpiler/vars.go [hunk 4]
```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/elastic/elastic-agent/internal/pkg/core/composable"
```
[the remark this answers is not in the document]
Comment (blakerouse) [reply internal/pkg/agent/transpiler/vars.go:14]: Then its wrong, it cannot be removed.

File: internal/pkg/core/status/handler.go [hunk 5]
```diff
@@ -0,0 +1,43 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package status
```
[the remark this answers is not in the document]
Comment (blakerouse) [reply internal/pkg/core/status/handler.go]: Yes I have moved this to `coordinator/handler.go`. So the `Coordinator` now as the `ServeHTTP` function that will return the overall status of Elastic Agent.

**Comment by blakerouse** [comment]
Lint is failing because of something that is in main, not something this PR changed or anything from feature-arch-v2.

Package building failed only do to out of space, everything else passes.

**Review by ph — APPROVED** [review]

**Pull request merged**